api/router/exec.go

```diff
@@ -0,0 +1,68 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/rs/zerolog/log"
+	"github.com/unweave/unweave/api/middleware"
+	"github.com/unweave/unweave/api/types"
+	execsrv "github.com/unweave/unweave/wip/services/exec"
+)
+
+type ExecRouter struct {
+	store execsrv.Store
+
+	llService        *execsrv.Service
+	unweaveService   *execsrv.Service
+	conductorService *execsrv.Service
+}
+
+func NewExecRouter(store execsrv.Store, lambdaLabsDriver, unweaveDriver execsrv.Driver) *ExecRouter {
+	lls, err := execsrv.NewService(store, lambdaLabsDriver)
+	if err != nil {
+		panic(fmt.Errorf("failed to create lambda labs service: %w", err))
+	}
+
+	return &ExecRouter{
+		store:            store,
+		llService:        lls,
+		unweaveService:   nil,
+		conductorService: nil,
+	}
+}
+
+func (e *ExecRouter) route(provider types.Provider) *execsrv.Service {
+	switch provider {
+	case types.LambdaLabsProvider:
+		return e.llService
+	case types.UnweaveProvider:
+		panic("not implemented") // this is only available in the hosted version
```

Comment:
Wouldn't it be more readable to register it in the map?
Btw. if you think in other way, please explain then we can learn from each other 😄